package/MDAnalysis/topology/TPRParser.py

```diff
@@ -226,6 +227,20 @@ def parse(self, **kwargs):
             raise IOError(msg)
 
         tpr_top.add_TopologyAttr(Resnums(tpr_top.resids.values.copy()))
+        
+        # Defining a way to obtain the elements data
+        atom_names = tpr_top.names.values.copy()
+        atom_mass  = tpr_top.masses.values.copy()
+        
+        if any(atom_names):
+            elements = guess_types(atom_names)
+        else:
+            warnings.warn("Guessing elements from atomic mass.")
+            elements = guess_types(guess_elements(masses))
```

Comment:
As previously discussed, https://github.com/MDAnalysis/mdanalysis/issues/2553#issuecomment-591823550, mass guessers tend to be rather messy for things like HMR. Whilst there would still be quite a few failure cases, in light of https://github.com/MDAnalysis/mdanalysis/issues/2553#issuecomment-594032676 an atomic_number guesser might be more appropriate (at least it would allow us to capture Martini CG atoms properly).